// crypto-extra/lib/index.js
'use strict';

var _crypto = require('crypto');

var _crypto2 = _interopRequireDefault(_crypto);

var _encryption = require('./encryption');

var _hash = require('./hash');

var _random = require('./random');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

module.exports = Object.assign(_crypto2.default, {
  generateKey: _encryption.generateKey,
  encrypt: _encryption.encrypt,
  decrypt: _encryption.decrypt,
  hash: _hash.hash,
  randomString: _random.randomString,
  randomNumber: _random.randomNumber,

  bcryptHash: function bcryptHash() {
    return deprecationNotice('bcrypt is no longer supported in this module! See https://github.com/jsonmaur/node-crypto-extra/issues/1');
  },
  bcryptCompare: function bcryptCompare() {
    return deprecationNotice('bcrypt is no longer supported in this module! See https://github.com/jsonmaur/node-crypto-extra/issues/1');
  }
});

function deprecationNotice(msg) {
  console.log('crypto-extra: ' + msg);
}
